plugins/outputs/opentsdb/opentsdb.go

```diff
@@ -136,10 +137,14 @@ func (o *OpenTSDB) WriteHttp(metrics []telegraf.Metric, u *url.URL) error {
 		tags := cleanTags(m.Tags())
 
 		for fieldName, value := range m.Fields() {
-			switch value.(type) {
+			switch fv := value.(type) {
 			case int64:
 			case uint64:
 			case float64:
+				// JSON does not support these special values
+				if math.IsNaN(fv) || math.IsInf(fv, 0) {
+					continue
```

Comment:
Looks like elsewhere these are serialized as `null`. should we be doing that instead?

Reply:
> 
> 
> Looks like elsewhere these are serialized as `null`. should we be doing that instead?

I do not think so. I dont found any of. So. Here we have a type checking, its logical to add checking of float64 for NaN or Inf. And we have errors for NaN from 1.14 version =)